harlog.go

```diff
@@ -49,6 +49,13 @@ func (proxy *ProxyHttpServer) harLogAggregator() {
 			}
 
 			proxy.harLog = har.New() // reset
+		case memory := <-proxy.HarFlushToMemory:
+			proxy.Logf("Received HAR flush request to memory")
+			if len(proxy.harLog.Log.Entries) == 0 {
+				proxy.Logf("No HAR entries to flush")
+				continue
+			}
+			memory <-proxy.harLog
```

Comment:
wouldn't you create a new harLog here then ? thus freeing the previous `proxy.harLog` for the consumer of `memory`.. otherwise, there are risks of two goroutines using the same `harLog`.